File: 7-reverse-integer/reverse-integer.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x>0 and x<2**31:
            y = int(str(x)[::-1])
            return y if y<(2**31) else 0
        elif x>=(-2**31):
            y = -1* int(str(x*-1)[::-1])
            return y if y>=(-2**31) else 0
        else:
            return 0


        

File: 7-reverse-integer/reverse-integer.java
class Solution {
    public int reverse(int x) {
        long ans=0;
        boolean flag =false;
        if(x<0)
        {
            flag = true;
            x = x*-1;
        }
        while(x>0)
        {
            ans = ans*10 + x%10;
            x = x/10;
        }
        if(flag)
        {
            ans = -1*ans;
        }
        return (Integer.MIN_VALUE<ans && Integer.MAX_VALUE>ans) ? (int)ans : 0;

    }
}